### PR TITLE
XCSS prop local cx lint

### PR DESCRIPTION
**Pull request by itsdouges**
This pull request adds a new lint rule `local-cx-xcss` that marks any usage of `cx()` outside of `xcss` prop as a violation.

I will be putting up small PRs over the week to complete all the lint rules we need for the xcss prop feature. The lint rules are defined here: https://github.com/atlassian-labs/compiled/pull/1533

### CODE REVIEW

File: packages/eslint-plugin/src/rules/local-cx-xcss/__tests__/rule.test.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      bg: { background: 'blue' },
+    });
+
+    const joinedStyles = cx(styles.text, styles.bg);
+
+    <Button xcss={styles} />;
+    `,
+    `
+    // Ignore cx usage not from compiled
+      const styles = cx({});
```
Comment (MonicaOlejniczak) [inline packages/eslint-plugin/src/rules/local-cx-xcss/__tests__/rule.test.ts:39]: /nit make the spacing consistent, so this line and the next should be lined up with the commend above

File: packages/eslint-plugin/src/rules/local-cx-xcss/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# `local-cx-xcss`
+
+This rule ensures the `cx()` function is only used within the `xcss` prop. This aids tracking what styles are applied to a jsx element.
+
+👎 Examples of **incorrect** code for this rule:
+
+```js
+import { cx, styleMap } from '@compiled/react';
```
Comment (MonicaOlejniczak) [inline packages/eslint-plugin/src/rules/local-cx-xcss/README.md:8]: `styleMap` is `cssMap` as per https://github.com/atlassian-labs/compiled/blob/master/packages/react/src/index.ts#L12

File: packages/eslint-plugin/src/rules/local-cx-xcss/__tests__/rule.test.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    <Component xcss={cx({})} />
+  `,
+    `
+    import { cx } from '@compiled/react';
+
+    <Component innerXcss={cx({})} />
+  `,
+    `
+    import { cx } from '@compiled/react';
```
Comment (MonicaOlejniczak) [inline packages/eslint-plugin/src/rules/local-cx-xcss/__tests__/rule.test.ts:17]: The `cssMap` import should be added for completeness

File: packages/eslint-plugin/src/rules/local-cx-xcss/README.md (hunk 2 above)
Comment (itsdouges) [reply packages/eslint-plugin/src/rules/local-cx-xcss/README.md:8]: Thanks will fix up

File: packages/eslint-plugin/src/rules/local-cx-xcss/__tests__/rule.test.ts (hunk 3 above)
Comment (itsdouges) [reply packages/eslint-plugin/src/rules/local-cx-xcss/__tests__/rule.test.ts:17]: Thanks

File: packages/eslint-plugin/src/rules/local-cx-xcss/__tests__/rule.test.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      import { cx } from '@compiled/react';
+
+      const styles = styleMap({
+        text: { color: 'red' },
+        bg: { background: 'blue' },
+      });
+
+      const joinedStyles = cx(styles.text, styles.bg);
+
+      <Button xcss={styles} />;
```
Comment (MonicaOlejniczak) [inline packages/eslint-plugin/src/rules/local-cx-xcss/__tests__/rule.test.ts:65]: Should be `joinedStyles` here and in other cases?

Comment (itsdouges) [reply packages/eslint-plugin/src/rules/local-cx-xcss/__tests__/rule.test.ts:65]: Yes. Will fix up.

**Review by JakeLane — APPROVED** [review]

**Review by MonicaOlejniczak — APPROVED** [review]

**Pull request merged**